src/grandorgue/sound/GOSound.cpp

```diff
@@ -319,7 +319,7 @@ bool GOSound::AudioCallback(
   GOSoundOutput *device = &m_AudioOutputs[dev_index];
   GOMutexLocker locker(device->mutex);
 
-  if (device->wait && device->waiting)
+  while (device->wait && device->waiting)
```

Comment:
Why did you change the behavior?

Reply:
From my understanding, this code should wait until the condition is met, e.g. you want to wait until https://github.com/GrandOrgue/grandorgue/blob/bf5144d9dc34d6ae4daa85c01fd9bec81d8526db/src/grandorgue/sound/GOSound.cpp#L194-L198 was executed. However, condition variables suffer from spurious wakeups (https://en.wikipedia.org/wiki/Spurious_wakeup) resulting in the function call returning before the condition is fulfilled. Therefore, conditions awaited using condition variables should always be checked in a loop.